src/main/java/io/unityfoundation/auth/AuthController.java

```diff
@@ -0,0 +1,88 @@
+package io.unityfoundation.auth;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.QueryValue;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.authentication.Authentication;
+import io.micronaut.security.rules.SecurityRule;
+import io.unityfoundation.auth.entities.Permission.PermissionScope;
+import io.unityfoundation.auth.entities.Service;
+import io.unityfoundation.auth.entities.ServiceRepo;
+import io.unityfoundation.auth.entities.Tenant;
+import io.unityfoundation.auth.entities.Tenant.TenantStatus;
+import io.unityfoundation.auth.entities.User;
+import io.unityfoundation.auth.entities.UserRepo;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/api")
+public class AuthController {
+
+  private final UserRepo userRepo;
+  private final ServiceRepo serviceRepo;
+
+  public AuthController(UserRepo userRepo, ServiceRepo serviceRepo) {
+    this.userRepo = userRepo;
+    this.serviceRepo = serviceRepo;
+  }
+
+  /**
+   * Checks if a user has permission based on the given request.
+   *
+   * @param requestDTO      the request object containing the necessary information
+   * @param subtenant       the flag indicating if the permission is for a subtenant (optional)
+   * @param authentication the user's authentication information
+   * @return true if the user has the permission, false otherwise
+   */
+  @Post("/hasPermission{?subtenant}")
+  public boolean hasPermission(@Body HasPermissionRequest requestDTO,
+      @Nullable @QueryValue("subtenant") Boolean subtenant, Authentication authentication) {
+    User user = userRepo.findByEmail(authentication.getName()).orElse(null);
+    if (user == null || user.getStatus() != User.UserStatus.ENABLED) {
+      return false;
+    }
+
+    if (requestDTO.tenantId() == null) {
+      return false;
+    }
+
+    Optional<Tenant> t = user.getTenants().stream()
```

Comment:
This won't work.  If I'm an auth admin at the SYSTEM level, then the tenant from the request will not be in `user.getTenants`.

Reply:
Logic assumes that there is tenant with name "SYSTEM". That way all logic is much more simple to implement.
The SYSTEM tenant is created with initialization script